app/models/paging_schedule.rb

```diff
@@ -31,26 +35,44 @@ def worst_case_delivery_day
 
     private
 
-    def schedule_for_request(request)
-      schedule.detect do |sched|
-        sched.from == request.origin &&
-          sched.to == request.destination &&
-          sched.by_time?(request.created_at)
-      end
+    def worst_case_days_later
+      schedule.filter_map(&:days_later).max + Settings.worst_case_paging_padding
     end
+  end
 
-    def default_schedule(request)
-      s = schedule.detect do |sched|
-        sched.from == request.origin &&
-          sched.to == :anywhere &&
-          sched.by_time?(request.created_at)
-      end
-      s&.for(request.destination)
+  def initialize(request)
+    @request = request
+  end
+
+  def schedule_or_default
+    schedule_for_destination || default_schedule || raise(ScheduleNotFound)
+  end
+
+  private
+
+  attr_reader :request
+
+  delegate :destination, :created_at, to: :request, prefix: 'target'
+
+  def target_origin
+    @target_origin ||= SAL3_COORDINATES.include?(request.origin) ? 'SAL3' : request.origin
```

Comment:
Isn't `request.origin` the library code?

Reply:
Yes. I'm not sure I see what you're hinting at though.

Reply:
`LANE-SAL3` and `LANE-SAL3X` are location codes. I'm not sure why we'd have a request with the location code in that attribute.

Reply:
Oh, I see my mistake here.